integration_test/src/test_random_miniscripts.rs

```diff
@@ -0,0 +1,57 @@
+//! # rust-miniscript integration test
+//!
+//! Randomly created miniscripts for fuzz testing against
+//! bitcoin core consensus rules
+
+use bitcoin;
+use bitcoin::hashes::hex::ToHex;
+use bitcoin::hashes::sha256;
+use miniscript::Miniscript;
+use miniscript::MiniscriptKey;
+use miniscript::Segwitv0;
+use std::fs::File;
+use std::io::{self, BufRead};
+use std::path::Path;
+use std::str::FromStr;
+
+type MsString = Miniscript<String, Segwitv0>;
+type Ms = Miniscript<bitcoin::PublicKey, Segwitv0>;
+
+// parse 1000 miniscripts from file
+pub fn parse_miniscripts(pks: &[bitcoin::PublicKey], hash: sha256::Hash) -> Vec<Ms> {
```

Comment:
should be an iterator rather than reading everything at once into a vec

Reply:
So, the tradeoff of avoiding allocations is that we will need to mine a single transaction per block to see that it is getting confirmed. If we collect all the miniscripts into a `Vec`. We can do the following. 
1) Create transactions spending to these Miniscripts.
2) Wait for confirmation
3) Spend from the Miniscripts
4) More confirmations

I guess I can mine 1 tx per block. Not a big deal.